test/integration/smoke/test_deploy_vm_root_resize.py

```diff
@@ -114,36 +134,46 @@ def test_00_deploy_vm_root_resize(self):
         # 2. root disk has new size per listVolumes
         # 3. Rejects non-supported hypervisor types
         """
-        if(self.hypervisor.lower() == 'kvm'):
-            newrootsize = (self.template.size >> 30) + 2
-            self.virtual_machine = VirtualMachine.create(
-                self.apiclient,
-                self.testdata["virtual_machine"],
-                accountid=self.account.name,
-                zoneid=self.zone.id,
-                domainid=self.account.domainid,
-                serviceofferingid=self.service_offering.id,
-                templateid=self.template.id,
-                rootdisksize=newrootsize
+
+
+        newrootsize = (self.template.size >> 30) + 2
+        if(self.hypervisor.lower() == 'kvm' or self.hypervisor.lower() ==
+            'xenserver'or self.hypervisor.lower() == 'vmware'  ):
+
+            if self.hypervisor=="vmware":
+                self.virtual_machine = VirtualMachine.create(
+                        self.apiclient, self.services["virtual_machine"],
+                        zoneid=self.zone.id,
+                        accountid=self.account.name,
+                        domainid=self.domain.id,
+                        serviceofferingid=self.services_offering_vmware.id,
+                        templateid=self.template.id
+                    )
+
```

Comment:
B.O. tests are failing because for vmware you don't specify rootdisksize=newrootsize. You probably better to remove if-else entirely.